src/types/routeMethods.test-d.ts

```diff
@@ -425,4 +425,19 @@ test('param names must be alphanumeric', () => {
     param3: string,
     param4: string,
   }>()
+})
+
+test('route method returns correct typ when called', async () => {
+  const routes = [
+    {
+      name: 'foo',
+      path: '/foo',
+      component,
+    },
+  ] as const satisfies Routes
+
+  const router = createRouter(routes)
+
+  expectTypeOf(router.routes.foo()).toMatchTypeOf<{ url: string }>()
```

Comment:
First time I'm using `toMatchTypeOf` which is a little looser than `toEqualTypeOf` because I could figure out how to match the `then`. So matching all but the `then` and then testing the thenable return type on the next line